diff --git a/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css b/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
index f83ad4b..fde5d2d 100644
--- a/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
+++ b/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
@@ -3,6 +3,9 @@
  * Copyright (C) 2019 The noVNC Authors
  * noVNC is licensed under the MPL 2.0 (see LICENSE.txt)
  * This file is licensed under the 2-Clause BSD license (see LICENSE.txt).
+ *
+ * Modified by b-data for seamless use with Jupyter Remote Desktop Proxy.
+ *
  */
 
 /*
@@ -630,7 +633,7 @@ html {
 }
 
 /* Connection Controls */
-:root:not(.noVNC_connected) #noVNC_disconnect_button {
+:root #noVNC_disconnect_button {
   display: none;
 }
 
diff --git a/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js b/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
index f27dfe2..f63386b 100644
--- a/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
+++ b/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
@@ -4,6 +4,9 @@
  * Licensed under MPL 2.0 (see LICENSE.txt)
  *
  * See README.md for usage and integration instructions.
+ * 
+ * Modified by b-data for seamless use with Jupyter Remote Desktop Proxy.
+ * 
  */
 
 import * as Log from '../core/util/logging.js';
@@ -120,7 +123,7 @@ const UI = {
 
         document.documentElement.classList.remove("noVNC_loading");
 
-        let autoconnect = WebUtil.getConfigVar('autoconnect', false);
+        let autoconnect = WebUtil.getConfigVar('autoconnect', true);
         if (autoconnect === 'true' || autoconnect == '1') {
             autoconnect = true;
             UI.connect();
@@ -174,15 +177,15 @@ const UI = {
         UI.initSetting('port', port);
         UI.initSetting('encrypt', (window.location.protocol === "https:"));
         UI.initSetting('view_clip', false);
-        UI.initSetting('resize', 'off');
+        UI.initSetting('resize', 'scale');
         UI.initSetting('quality', 6);
         UI.initSetting('compression', 2);
         UI.initSetting('shared', true);
         UI.initSetting('view_only', false);
         UI.initSetting('show_dot', false);
-        UI.initSetting('path', 'websockify');
+        UI.initSetting('path', window.location.pathname.replace(/[^/]*$/, '').substring(1) + 'websockify');
         UI.initSetting('repeaterID', '');
-        UI.initSetting('reconnect', false);
+        UI.initSetting('reconnect', true);
         UI.initSetting('reconnect_delay', 5000);
 
         UI.setupSettingLabels();
