diff --git a/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css b/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
index 87bfb45..b84e015 100644
--- a/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
+++ b/jupyter_remote_desktop_proxy/share/web/noVNC/app/styles/base.css
@@ -3,6 +3,9 @@
  * Copyright (C) 2019 The noVNC authors
  * noVNC is licensed under the MPL 2.0 (see LICENSE.txt)
  * This file is licensed under the 2-Clause BSD license (see LICENSE.txt).
+ *
+ * Modified by b-data for seamless use with Jupyter Remote Desktop Proxy.
+ *
  */
 
 /*
@@ -480,8 +483,6 @@ html {
 .noVNC_panel textarea,
 .noVNC_panel input:not([type=checkbox]):not([type=radio]) {
     margin-left: 6px;
-    /* Prevent inputs in panels from being too wide */
-    max-width: calc(100% - 6px - var(--input-xpadding) * 2);
 }
 
 .noVNC_panel .noVNC_heading {
@@ -639,7 +640,7 @@ html {
 }
 
 /* Connection controls */
-:root:not(.noVNC_connected) #noVNC_disconnect_button {
+:root #noVNC_disconnect_button {
     display: none;
 }
 
diff --git a/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js b/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
index 51e57bd..e2e12b6 100644
--- a/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
+++ b/jupyter_remote_desktop_proxy/share/web/noVNC/app/ui.js
@@ -4,6 +4,9 @@
  * Licensed under MPL 2.0 (see LICENSE.txt)
  *
  * See README.md for usage and integration instructions.
+ *
+ * Modified by b-data for seamless use with Jupyter Remote Desktop Proxy.
+ *
  */
 
 import * as Log from '../core/util/logging.js';
@@ -176,18 +179,18 @@ const UI = {
         UI.initSetting('port', 0);
         UI.initSetting('encrypt', (window.location.protocol === "https:"));
         UI.initSetting('password');
-        UI.initSetting('autoconnect', false);
+        UI.initSetting('autoconnect', true);
         UI.initSetting('view_clip', false);
-        UI.initSetting('resize', 'off');
+        UI.initSetting('resize', 'scale');
         UI.initSetting('quality', 6);
         UI.initSetting('compression', 2);
         UI.initSetting('shared', true);
         UI.initSetting('bell', 'on');
         UI.initSetting('view_only', false);
         UI.initSetting('show_dot', false);
-        UI.initSetting('path', 'websockify');
+        UI.initSetting('path', window.location.pathname.replace(/[^/]*$/, '').substring(1) + 'websockify');
         UI.initSetting('repeaterID', '');
-        UI.initSetting('reconnect', false);
+        UI.initSetting('reconnect', true);
         UI.initSetting('reconnect_delay', 5000);
     },
     // Adds a link to the label elements on the corresponding input elements
